fapolicy_analyzer/ui/features/daemon_feature.py

```diff
@@ -0,0 +1,179 @@
+import gi
+import os
+import logging
+import sys
+from time import sleep
+from threading import Thread
+
+gi.require_version("Gtk", "3.0")
+from fapolicy_analyzer import (
+    Handle,
+    is_fapolicyd_active,
+    start_fapolicyd,
+    stop_fapolicyd,
+)
+
+from fapolicy_analyzer.ui.actions import (
+    REQUEST_DAEMON_START,
+    REQUEST_DAEMON_STOP,
+    REQUEST_DAEMON_RELOAD,
+    REQUEST_DAEMON_STATUS,
+    REQUEST_DAEMON_STATUS_UPDATE,
+    error_daemon_reload,
+    error_daemon_start,
+    error_daemon_status,
+    error_daemon_stop,
+    init_daemon,
+    received_daemon_reload,
+    received_daemon_start,
+    received_daemon_status,
+    received_daemon_status_update,
+    received_daemon_stop,
+)
+from fapolicy_analyzer.ui.reducers import daemon_reducer
+from gi.repository import GLib
+from redux import (
+    Action,
+    combine_epics,
+    create_feature_module,
+    of_init_feature,
+    of_type,
+    ReduxFeatureModule,
+)
+from fapolicy_analyzer.ui.strings import DAEMON_INITIALIZATION_ERROR
+from rx import of, pipe
+from rx.operators import catch, map
+from typing import Callable
+
+DAEMON_FEATURE = "daemon"
+_fapd_status: bool = False
+_fapd_ref: Handle = None
+
+
+def create_daemon_feature(dispatch: Callable, daemon=None) -> ReduxFeatureModule:
+    """
+    Creates a Redux feature of type Daemon
+
+    Keyword arguments:
+    dispatch -- the Redux Store dispatch method, used for dispatching actions
+    daemon -- the fapolicy_analyzer.daemon object, defaults to None. If not
+    provided, a new Daemon object will be initialized.  Used for testing
+    purposes only.
+    """
+    def _init_daemon() -> Action:
+        logging.debug("_init_daemon() -> Action")
+
+        def acquire_daemon():
+            logging.debug("acquire_daemon()")
+            try:
+                daemon = Handle("fapolicyd")
```

Comment:
Where does this guy ever get used?  Do you need to initialize a `Handle` before you can use any of the other ' is_fapolicyd_active, start_fapolicyd, stop_fapolicyd' methods?

Reply:
The Handle does not currently get used.

Reply:
The handle/reference is now used for all interfacing to the backend for fapd status/control.